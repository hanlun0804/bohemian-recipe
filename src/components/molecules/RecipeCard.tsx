import React from 'react'

type RecipeCardProps = {
    imgSrc: string
    imgAlt: string
    title: string
    category: string
    onClick?: () => void // Add onClick prop to the props type
}

const RecipeCard: React.FC<RecipeCardProps> = ({
    imgSrc,
    imgAlt,
    title,
    category,
    onClick,
}) => {
    return (
        <div
            className="w-full h-96 rounded-md border-8 border-black bg-zinc-900 border-color-orange"
            onClick={onClick}
            style={{ cursor: 'pointer' }}
        >
            <img
                src={imgSrc}
                alt={imgAlt}
                className="rounded-t-md w-full h-1/2 object-cover"
            />

            <div className="p-4 text-white">
                <h2 className="text-xl font-semibold mb-2">{title}</h2>
                <hr className="my-2" />
                <p className="mt-auto">{category}</p>
            </div>
        </div>
    )
}

export default RecipeCard
