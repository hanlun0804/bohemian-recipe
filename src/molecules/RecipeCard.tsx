import React from 'react'
import './RecipeCard.css'

type RecipeCardProps = {
    imgSrc: string
    imgAlt: string
    title: string
}

const RecipeCard: React.FC<RecipeCardProps> = ({ imgSrc, imgAlt, title }) => {
    return (
        <div>
            <p>Hei</p>
            {imgSrc}
            {imgAlt}
            {title}
        </div>
    )
}

export default RecipeCard
