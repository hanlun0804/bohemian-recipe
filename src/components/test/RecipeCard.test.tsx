import { render, screen } from '@testing-library/react';
import {  expect, it } from 'vitest';
import RecipeCard from '../molecules/RecipeCard';

it('renders cards without error, with correct category, and correct title is displayed on card', async () => {
    render(
        <RecipeCard
            mealId="52971"
            strMealThumb="https://www.themealdb.com/images/media/meals/1bsv1q1560459826.jpg"
            title="Kafteji"
            strCategory="Vegetarian"
            rating={[5, 5, 2]}
            tabIndex={0}
        />
    );

    const recipeCard = screen.getByTestId('cypress-recipe-card-Vegetarian');
    const kafejiCard = screen.getByText('Kafteji');
    expect(recipeCard).toBeTruthy();
    expect(kafejiCard)
    expect(document.body).toMatchSnapshot()
})