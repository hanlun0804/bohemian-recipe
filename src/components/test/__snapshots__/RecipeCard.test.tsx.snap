// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders cards without error, with correct category, and correct title is displayed on card 1`] = `
<body>
  <div>
    <div
      class="w-full h-96 p-0 text-left rounded-xl border-8 border-black bg-zinc-900 border-color-orange transition-none"
      data-testid="cypress-recipe-card-Vegetarian"
      style="cursor: pointer;"
      tabindex="0"
      title="Kafteji"
    >
      <img
        class="rounded-t-md w-full h-1/2 object-cover"
        src="https://www.themealdb.com/images/media/meals/1bsv1q1560459826.jpg"
      />
      <div
        class="p-4 text-white"
      >
        <h2
          class="text-xl font-semibold mb-2"
        >
          Kafteji
        </h2>
        <hr
          class="my-2"
        />
        <p
          class="mt-auto"
          data-testid="cypress-recipe-category"
        >
          Vegetarian
        </p>
        <p
          class="mt-auto"
        >
          <span
            class=""
          >
            â˜…
          </span>
           
          4.00
           / 5
        </p>
      </div>
    </div>
  </div>
</body>
`;
